plotly/figure_factory/_bullet.py

```diff
@@ -0,0 +1,342 @@
+        in each subplot such as name and marker size. Call
+        help(plotly.graph_objs.Scatter) for more information on valid params.
+    :param layout_options: describes attributes for the layout of the figure
+        such as title, height and width. Call help(plotly.graph_objs.Layout)
+        for more information on valid params.
+
+    Example 1: Use a Dictionary
+    ```
+    import plotly
+    import plotly.plotly as py
+    import plotly.figure_factory as ff
+
+    data = [
+      {"e": "Revenue", "d": "US$, in thousands", "c": [150, 225, 300],
+       "b": [220,270], "a": [250]},
+      {"e": "Profit", "d": "%", "c": [20, 25, 30], "b": [21, 23], "a": [26]},
+      {"e": "Order Size", "d":"US$, average","c": [350, 500, 600],
+       "b": [100,320],"a": [550]},
+      {"e": "New Customers", "d": "count", "c": [1400, 2000, 2500],
+       "b": [1000,1650],"a": [2100]},
+      {"e": "Satisfaction", "d": "out of 5","c": [3.5, 4.25, 5],
+       "b": [3.2,4.7], "a": [4.4]}
+    ]
+
+    fig = ff.create_bullet(
+        data, titles='e', subtitles='d', markers='a', measures='b',
+        ranges='c', orientation='h', title='my simple bullet chart'
+    )
+    py.iplot(fig)
+    ```
+
+    Example 2: Use a DataFrame with Custom Colors
+    ```
+    import plotly.plotly as py
+    import plotly.figure_factory as ff
+
+    import pandas as pd
+
+    data = pd.read_json('https://cdn.rawgit.com/plotly/datasets/master/BulletData.json')
+
+    fig = ff.create_bullet(
+        data, titles='title', markers='markers', measures='measures',
+        orientation='v', measure_colors=['rgb(14, 52, 75)', 'rgb(31, 141, 127)'],
+        scatter_options={'marker': {'symbol': 'circle'}}, width=700
+
+    )
+    py.iplot(fig)
+    ```
+    """
+    # validate df
+    if not pd:
+        raise exceptions.ImportError(
+            "'pandas' must be imported for this figure_factory."
+        )
+
+    if isinstance(data, list):
+        if not all(isinstance(item, dict) for item in data):
+            raise exceptions.PlotlyError(
+                'If your data is a list, all entries must be dictionaries.'
+            )
+
+    elif not isinstance(data, pd.DataFrame):
+        raise exceptions.PlotlyError(
+            'You must input a pandas DataFrame or a list of dictionaries.'
+        )
+
+    # make DataFrame from data with correct column headers
+    col_names = ['titles', 'subtitle', 'markers', 'measures', 'ranges']
+    if isinstance(data, list):
+        df = pd.DataFrame(
+            [
+                [d[titles] for d in data] if titles else [''] * len(data),
+                [d[subtitles] for d in data] if subtitles else [''] * len(data),
+                [d[markers] for d in data] if markers else [[]] * len(data),
+                [d[measures] for d in data] if measures else [[]] * len(data),
+                [d[ranges] for d in data] if ranges else [[]] * len(data),
+            ],
+            index=col_names
```

Comment:
Rather than looping through every element and reconstructing a list, I believe that you can just pass in your list of dictionaries directly into pandas:
```
In [2]: pd.DataFrame([{'x': 1, 'y': 3}, {'x': 3, 'y': 10}])
Out[2]:
   x   y
0  1   3
1  3  10
```

Reply:
That's true, but I want to insert empty columns where a column name is not inserted.

Reply:
Ex. if the user doesn't assign the `titles` variable to anything, I want a column named `titles` with empty strings in each cell of that column

Reply:
Can I leave this the same way it is? I think it will be more straightforward than doing `pd.DataFrame(data)` then adding in the empty columns afterwards.

Reply:
sure